### PR TITLE
RBAC add missing resource to kubebuilder set

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

#### What this PR does / why we need it:
This is on the logs 
```bash
1.6455346824721851e+09  LEVEL(-2)       job-reconciler  Job assigned a capacity, unsuspending{"job": {"name":"sample-job-zkvlc","namespace":"default"}}                                       
E0222 12:58:02.473038       1 event.go:267] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"sample-job-zkvlc.16d61d13464ee05b", 
GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, D
eletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:""
, ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"QueuedWorkload", Namespace:"default", Name:"sample-job-zkvlc", UID:"b495fad9-c39c-49b7-8200-62a575326ce
a", APIVersion:"kueue.x-k8s.io/v1alpha1", ResourceVersion:"34819", FieldPath:""}, Reason:"Assigned", Message:"Assigned to capacity cluster-total", Source:v1.EventSource{Component:"kueue-mana
ger", Host:""}, FirstTimestamp:time.Date(2022, time.February, 22, 12, 58, 2, 471653467, time.Local), LastTimestamp:time.Date(2022, time.February, 22, 12, 58, 2, 471653467, time.Local), Count
:1, Type:"Normal", EventTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", Report
ingInstance:""}': 'events is forbidden: User "system:serviceaccount:kueue-system:kueue-controller-manager" cannot create resource "events" in API group "" in the namespace "default"' (will n
ot retry!)  
```

TLDR:  ` 'events is forbidden: User "system:serviceaccount:kueue-system:kueue-controller-manager" cannot create resource "events" in API group "" in the namespace "default"' `

THis patch fix the underlying issue, now the logs are clean, and RBAC is happy

### CODE REVIEW

File: pkg/controller/workload/job/job_controller.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ func (r *JobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;watch;update
```
Comment (ahg-g) [inline pkg/controller/workload/job/job_controller.go:86]: can we also add this to the core controllers as well just in case this controller gets refactored out?

Comment (ArangoGutierrez) [reply pkg/controller/workload/job/job_controller.go:86]: well we can do a diff thing, we can add this to kustomize , example
```yaml
patchesJSON6902:
- target:
    kind: ClusterRole
    name: manager-role
  patch: |-
    - op: add
      path: /rules/0
      value:
        apiGroups:
        - ""
        resources:
        - events
        verbs:
        - create
        - update
  ```

Comment (ArangoGutierrez) [reply pkg/controller/workload/job/job_controller.go:86]: this way we don't worry about refactors in the go code, and kustomize keeps us in place

Comment (ahg-g) [reply pkg/controller/workload/job/job_controller.go:86]: wouldn't jus adding this line to the other controllers be simpler? can we add it to the [main.go](https://github.com/kubernetes-sigs/kueue/blob/main/main.go) file for example or does it have to be in the one of the controllers files?

Comment (ArangoGutierrez) [reply pkg/controller/workload/job/job_controller.go:86]: did to the controllers as initially requested

**Comment by alculquicondor** [comment]
it sounds like we need minimal E2E tests

**Comment by ahg-g** [comment]
I thought integration tests cover RBAC!

**Comment by alculquicondor** [comment]
no.... only the CRDs
